Controllers/ProductController.cs

```diff
@@ -0,0 +1,130 @@
+﻿using EFCoreServices.Models;
+using EFCoreServices.Repository;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace EFCoreServices.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProductController : ControllerBase
+    {
+        IProductRepository productRepository;
+        public ProductController(ProductRepository _productRepository)
+        {
+            productRepository = _productRepository;
+        }
+
+        [HttpGet]
+        [Route("GetProductsList")]
+        public async Task<IActionResult> GetProductsList()
+        {
+            try
+            {
+                var products = await productRepository.GetProductsList();
+                if(products == null)
+                {
+                    return NotFound();
```

Comment:
generally, we dont return not found on when requesting lists, its more for specific items like get by id or something 
you can use NoContent when there is not data.